table/block_based/block_prefetcher.cc

```diff
@@ -16,17 +16,29 @@ void BlockPrefetcher::PrefetchIfNeeded(const BlockBasedTable::Rep* rep,
   if (is_for_compaction) {
     rep->CreateFilePrefetchBufferIfNotExists(compaction_readahead_size_,
                                              compaction_readahead_size_,
-                                             &prefetch_buffer_);
+                                             &prefetch_buffer_, false);
     return;
   }
 
   // Explicit user requested readahead
   if (readahead_size > 0) {
     rep->CreateFilePrefetchBufferIfNotExists(readahead_size, readahead_size,
-                                             &prefetch_buffer_);
+                                             &prefetch_buffer_, false);
     return;
   }
 
+  // Enable implicit auto readahead if reads are sequential based on prev
+  // offset/length read and num_file_reads_.
+  if (prev_len_ != 0 && (prev_offset_ > handle.offset() ||
```

Comment:
I wonder if this might be too strict. Perhaps we should also consider if the required data is satisfied by the prefetch buffer, even if its not strictly sequential.

Reply:
@anand1976, I am sorry I didn't get what do you mean by "required data is satisfied by the prefetch buffer". Which conditions are you refering to?

Reply:
I mean if ```handle.offset() + block_size(handle) <= readahead_limit_```, then even though the block referred by ```handle``` is not strictly sequential, the previous readahead was still useful so it might make sense to not reset.